app/views/reservations.js

```diff
@@ -51,13 +51,17 @@ return  ['$scope','$document','mongoStorage','ngDialog','$rootScope','$timeout',
       $scope.$watch('reservationsCtrl.sort',function(){
         getReservations();
       });
+      $scope.$watch('reservationsCtrl.conference._id',function(){
```

Comment:
for future, try removing the watch after init as init will not be called again in the lifecycle